[CI] AutoFollowIT midnight timing related failures

**Build scan**: https://gradle-enterprise.elastic.co/s/btcm27tw2kzqk

**Repro line**: `./gradlew ':x-pack:plugin:ccr:qa:multi-cluster:follow-cluster' --tests "org.elasticsearch.xpack.ccr.AutoFollowIT.testDataStreams_autoFollowAfterDataStreamCreated" -Dtests.seed=AFE0A472E2813558 -Dtests.security.manager=true -Dtests.locale=be-BY -Dtests.timezone=Africa/Maseru -Druntime.java=11`

and: `./gradlew ':x-pack:plugin:ccr:qa:multi-cluster:follow-cluster' --tests "org.elasticsearch.xpack.ccr.AutoFollowIT.testDataStreamsBiDirectionalReplication" -Dtests.seed=AFE0A472E2813558 -Dtests.security.manager=true -Dtests.locale=be-BY -Dtests.timezone=Africa/Maseru -Druntime.java=11`

**Reproduces locally?**: Not tried yet.

**Applicable branches**: Currently master, but this test should fail around midnight as well.

**Failure history**: [at least 4 failures in the last 7 days](https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=Europe/Amsterdam&tests.container=org.elasticsearch.xpack.ccr.AutoFollowIT&tests.sortField=FAILED&tests.test=testDataStreamsBiDirectionalReplication&tests.unstableOnly=true). 

**Failure excerpt**:

```
java.lang.AssertionError: |  
-- | --
  | Expected: ".ds-logs-http-eu-2021.04.02-000001" |  
  | but: was ".ds-logs-http-eu-2021.04.01-000001" |  
at __randomizedtesting.SeedInfo.seed([AFE0A472E2813558:5D547772E9490053]:0) |  
-- | --
  |   | •••
  |   | at org.elasticsearch.xpack.ccr.ESCCRRestTestCase.verifyDataStream(ESCCRRestTestCase.java:319) |  
  |   | at org.elasticsearch.xpack.ccr.AutoFollowIT.testDataStreamsBiDirectionalReplication(AutoFollowIT.java:581)
```

The tests shouldn't do an equal check on the name of the backing index. Instead it should check that the backing index name has the expected structure, data stream name and generation.

--- a/x-pack/plugin/ccr/qa/src/main/java/org/elasticsearch/xpack/ccr/ESCCRRestTestCase.java
+++ b/x-pack/plugin/ccr/qa/src/main/java/org/elasticsearch/xpack/ccr/ESCCRRestTestCase.java
@@ -311,12 +311,20 @@ public class ESCCRRestTestCase extends ESRestTestCase {
         Map<String, ?> response = toMap(client.performRequest(request));
         List<?> retrievedDataStreams = (List<?>) response.get("data_streams");
         assertThat(retrievedDataStreams, hasSize(1));
-        List<?> actualBackingIndices = (List<?>) ((Map<?, ?>) retrievedDataStreams.get(0)).get("indices");
-        assertThat(actualBackingIndices, hasSize(expectedBackingIndices.length));
+        List<?> actualBackingIndexItems = (List<?>) ((Map<?, ?>) retrievedDataStreams.get(0)).get("indices");
+        assertThat(actualBackingIndexItems, hasSize(expectedBackingIndices.length));
         for (int i = 0; i < expectedBackingIndices.length; i++) {
-            Map<?, ?> actualBackingIndex = (Map<?, ?>) actualBackingIndices.get(i);
+            Map<?, ?> actualBackingIndexItem = (Map<?, ?>) actualBackingIndexItems.get(i);
+            String actualBackingIndex = (String) actualBackingIndexItem.get("index_name");
             String expectedBackingIndex = expectedBackingIndices[i];
-            assertThat(actualBackingIndex.get("index_name"), equalTo(expectedBackingIndex));
+
+            String actualDataStreamName = actualBackingIndex.substring(5, actualBackingIndex.indexOf('-', 5));
+            String expectedDataStreamName = expectedBackingIndex.substring(5, expectedBackingIndex.indexOf('-', 5));
+            assertThat(actualDataStreamName, equalTo(expectedDataStreamName));
+
+            int actualGeneration = Integer.parseInt(actualBackingIndex.substring(actualBackingIndex.lastIndexOf('-')));
+            int expectedGeneration = Integer.parseInt(expectedBackingIndex.substring(expectedBackingIndex.lastIndexOf('-')));
+            assertThat(actualGeneration, equalTo(expectedGeneration));
         }
     }
 

